osc/fetch.py

```diff
@@ -258,19 +268,26 @@ def run(self, buildinfo):
                         # mark it for downloading from the API
                         self.__add_cpio(i)
                     else:
-                        # if the checksum of the downloaded package doesn't match,
-                        # delete it and mark it for downloading from the API
+                        # HDRMD5 MISMATCH EXPLANATION:
+                        # OBS builds the noarch packages once per binary arch.
+                        # Such noarch packages are supposed to be nearly identical across all build arches,
+                        # any discrepancy in the payload and dependencies is considered a packaging bug.
+                        # But to guarantee that the local builds work identically to builds in OBS,
+                        # using the arch-specific copy of the noarch package is required.
+                        # Unfortunatelly only one of the noarch packages gets distributed
+                        # and can be downloaded from a local mirror.
+                        # All other noarch packages are available through the OBS API only.
+                        # Since there is currently no information about hdrmd5 checksums of published noarch packages,
+                        # we download them, verify hdrmd5 and re-download the package from OBS API on mismatch.
                         #
-                        # wbrown 2022 - is there a reason to keep these md5's at all? md5 is
-                        # broken from a security POV so these aren't a trusted source for validation
-                        # of the file content. They are often incorrect forcing download via the API
-                        # which for anyone outside the EU is excruciating. And when they are ignored
-                        # builds work and progress anyway? So what do they even do? What are they
-                        # for? They should just be removed.
+                        # FIXME: We're currently ignoring the hdrmd5 verification of rpm packages.
+                        # A proper fix requires OBS change such as described in
+                        # https://github.com/openSUSE/open-build-service/issues/13821
```

Comment:
please do not ignore it by default. The default should be really safe and reproduce the server side state as highest goal.

An optional switch to disable it should be implemented within a minute ....

Reply:
Yes, the `disable_hdrmd5_check` option will default to `0` (the original behavior when the exact checksums were required)